sdk/communication/Azure.Communication.CallAutomation/src/CustomBuildHttpPieline.cs

```diff
@@ -14,7 +14,9 @@ public static HttpPipeline CustomBuildHttpPipeline(this ClientOptions options, C
             string stringSign = new Uri(connectionString.GetRequired("endpoint")).Host;
 
             var authPolicy = new CustomHMACAuthenticationPolicy(new AzureKeyCredential(connectionString.GetRequired("accesskey")), stringSign);
-            return HttpPipelineBuilder.Build(options, authPolicy);
+            HttpPipelineOptions httpPipelineOptions = new(options) { PerRetryPolicies = { authPolicy } };
+            HttpPipelineTransportOptions httpPipelineTransportOptions = new() { IsClientRedirectEnabled = true };
```

Comment:
Could this use options.ClientRedirects.IsClientRedirectEnabled instead?

Reply:
Ah, gotcha, yes that makes sense.
I think I was misunderstanding that HttpPipelineTransportOptions isn't also on ClientOptions.

Can you help me understand why we have HttpPipelineTransportOptions in the public API?  Sorry that I don't remember this well.  Is it only for library authors, and not end-user customers?

So then we have a precedence ordering of:
1. Per message: RedirectPolicy.SetAllowAutoRedirect(message, true);
2. Per client, set by end-user customer: ClientOptions.ClientRedirects.IsClientRedirectEnabled = true;
3. Per client, set by library author: HttpPipelineTransportOptions.IsClientRedirectEnabled = true;

Where a lower number will override a higher number on that list?

To [Josh's point](https://github.com/Azure/azure-sdk-for-net/pull/34645#issuecomment-1581396095), do we have a need for item 2 on the list?  It seems like we could always add that in response to customer feedback, but not providing it now improves security of our libraries.

Reply:
Removing the clientOptions API.